src/next_state/stack.rs

```diff
@@ -211,3 +211,113 @@ pub trait NextStateStackMutExtClone: NextStateStackMut + Clone {
 }
 
 impl<S: NextStateStackMut + Clone> NextStateStackMutExtClone for S {}
+
+/// An extension trait for [`Commands`] that provides methods for operating on states with
+/// [`NextStateStack`] as their `Next` type.
+pub trait NextStateStackCommandsExt {
+    /// Push a new base state index to the stack.
+    fn state_stack_acquire<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self;
+
+    /// Pop the top base state index of the stack.
+    fn state_stack_release<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self;
+
+    /// Clear the stack down to the base state.
+    fn state_stack_clear<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self;
+
+    /// Pop the stack if it's above the base state.
+    fn state_stack_pop<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self;
+
+    /// Push a state to the top of the stack.
+    fn state_stack_push<S: State<Next = NextStateStack<S>>>(&mut self, state: S) -> &mut Self;
+
+    /// Clear and then push a state to the top of the stack.
+    fn state_stack_clear_push<S: State<Next = NextStateStack<S>>>(&mut self, state: S)
+        -> &mut Self;
+
+    /// Pop and then push a state to the top of the stack.
+    fn state_stack_pop_push<S: State<Next = NextStateStack<S>>>(&mut self, state: S) -> &mut Self;
+}
+
+impl NextStateStackCommandsExt for Commands<'_, '_> {
+    fn state_stack_acquire<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self {
+        self.queue(state_stack_acquire::<S>);
+        self
+    }
+
+    fn state_stack_release<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self {
+        self.queue(state_stack_release::<S>);
+        self
+    }
+
+    fn state_stack_clear<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self {
+        self.queue(state_stack_clear::<S>);
+        self
+    }
+
+    fn state_stack_pop<S: State<Next = NextStateStack<S>>>(&mut self) -> &mut Self {
+        self.queue(state_stack_pop::<S>);
+        self
+    }
+
+    fn state_stack_push<S: State<Next = NextStateStack<S>>>(&mut self, state: S) -> &mut Self {
+        self.queue(state_stack_push(state));
+        self
+    }
+
+    fn state_stack_clear_push<S: State<Next = NextStateStack<S>>>(
+        &mut self,
+        state: S,
+    ) -> &mut Self {
+        self.queue(state_stack_clear_push(state));
+        self
+    }
+
+    fn state_stack_pop_push<S: State<Next = NextStateStack<S>>>(&mut self, state: S) -> &mut Self {
+        self.queue(state_stack_pop_push(state));
+        self
+    }
+}
+
+/// [`Command`] impl for [`NextStateStackCommandsExt::state_stack_acquire`].
+pub fn state_stack_acquire<S: State<Next = NextStateStack<S>>>(world: &mut World) {
```

Comment:
```suggestion
fn state_stack_acquire<S: State<Next = NextStateStack<S>>>(world: &mut World) {
```
Is there a reason for this and the other functions to be `pub`? Also, is there a reason for them to be named functions instead of closures?

Reply:
I think it's mostly preference, but I know there are cases when you want to be able to name the command you're using. You can always wrap it though, so I'm fine to change it!